arch/arm/core/aarch64/Kconfig

```diff
@@ -90,6 +90,83 @@ config GEN_ISR_TABLES
 config GEN_IRQ_VECTOR_TABLE
 	default n
 
+config ARM_MMU
+	bool "ARM MMU Support"
+	help
+	  Memory Management Unit support.
```

Comment:
What are the (reasonable) dependencies here?
E.g. for Cortex-M the MPU is an optional Cortex peripheral, so the dependency is CPU_HAS_MPU.
What is the case for Cortex-A, is MMU a mandatory feature?

Also, does it make sense to build the core without MMU support if this is present? If not, this could be default y.

Reply:
>>Does it make sense to build the core without MMU support if this is present ?

Cortex-A (v8/v7) MMU Hardware is mandatory feature. Sometimes for debug purpose it might make sense to build the code without MMU even if support is present.

Reply:
> Also, does it make sense to build the core without MMU support if this is present? If not, this could be default y.

It should definitely be `default y`. Unaligned accesses are not supported (RAM is treated as Device-nGnRnE) when the MMU is disabled and this means you will be seeing lots of alignment faults in real hardware.

Reply:
We will enable by default. If required for debug purpose, one can manually modify `Kconfig` and disable MMU.
Note that till now we were adding `select ARM_MMU` in board config(ex.: `soc/arm/qemu_cortex_a53/Kconfig.soc`), will remove that.

Reply:
ok good